File: src/code/java/Q268.java
package code.java;

public class Q268 {
    /*
    public int missingNumber(int[] nums) {
        int sum = nums.length * (nums.length + 1) / 2;
        for (int num : nums) {
            sum -= num;
        }
        return sum;
    }
    */
    public int missingNumber(int[] nums) {
        int n = nums.length;
        for (int i = 0; i < nums.length; i++) {
            // 边加边减 防止溢出
            n += (i - nums[i]);
        }
        return n;
    }
}

File: src/code/js/Q268.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
    var sum = nums.length * (nums.length + 1) / 2;
    for (var i = 0; i < nums.length; i++) {
        sum -= nums[i];
    }
    return sum;
};